fix: reset backend connection when client abort sse request

jhaeyaert opened the pull request:
## Issue

https://gravitee.atlassian.net/browse/APIM-11575
https://gravitee.atlassian.net/browse/APIM-10919

## Description

**This replaces previous [PR](https://github.com/gravitee-io/gravitee-api-management/pull/13976). Generated Docker image tag was too long to be deployed**

This PR fixes an issue when using a V4 HTTP proxy API or a V2 API with V4 emulation engine to perform SSE streaming (e.g., backend does SSE). 
SSE streaming is subject to never-ending. When the client was aborting the SSE stream, the gateway was unable to detect it and continued to consume the backend response under the hood. This led to resource exhaustion (CPU, Memory, ...).

The PR fixes the issue and introduces several logging improvements as well.

## Additional information

Tests have been conducted to validate that SSE requests still work the same for both V4 HTTP Proxy and V4 Protocol Mediation (SSE entrypoint). 

**For V4 Protocol Mediation**: the upstream (Kafka, MQTT, Mock, ...) stays properly cancelled when the client aborts the SSE stream.

**For V4 HTTP Proxy**: the upstream request is reset when the client aborts the SSE stream. This has been tested with both HTTP/1.1 and HTTP/2 with and without Keep Alive enabled: 
- HTTP1 Keep Alive disabled -> OK, connection is closed if SSE ends normally or is cancelled by the client
- HTTP1 Keep Alive enabled -> OK, connection kept if SSE ends normally (ex, finite stream), connection is reset and closed if SSE is cancelled by the client
- HTTP2 (note: there is no support for Keep alive, UI should not provide this option) -> OK, connection is kept if SSE ends normally, connection is also kept if SSE is cancelled by the client, but the HTTP2 stream is properly cancelled.

Tagging @jgiovaresco, @michel-barret, and @phiz71 as HttpConnector is also used in other areas in your scope (llm, mcp).

jhaeyaert committed:
fix: reset backend connection when client abort sse request

https://gravitee.atlassian.net/browse/APIM-11575

jgiovaresco reviewed (approved):
LGTM, Can we add the API id in the log we added? I don't if it is possible but having the API id would ease the debuging 🤔

jhaeyaert commented:
> LGTM, Can we add the API id in the log we added? I don't if it is possible but having the API id would ease the debuging 🤔

I had a look at how it is done and found several places where such information is missing. I could add it specifically on the few logs I've added, but I feel like it should be done in a dedicated task to improve the readability globally. I'll open a ticket for that.

jhaeyaert committed:
chore: update gravitee-node to 7.17.1

jhaeyaert commented:
@gravitee-io/platform can you approve 🙏
